docs/src/modules/components/AppFrame.js

```diff
@@ -34,6 +34,7 @@ import { LANGUAGES_LABEL } from 'docs/src/modules/constants';
 import { pathnameToLanguage } from 'docs/src/modules/utils/helpers';
 import { useChangeTheme } from 'docs/src/modules/components/ThemeContext';
 import PageContext from 'docs/src/modules/components/PageContext';
+import Skeleton from '@material-ui/lab/Skeleton';
```

Comment:
I would be in favor of not using the skeleton at all for these changes. I think that the usage of this component should be limited to the main elements of the page. In our case, these are relatively details that don't represent the most important aspect of the page.

Reply:
For the search box, my preference would be to render it on the server, but to reconcile what the user might have type inside the field once the input comes interactive.

Reply:
> For the search box, my preference would be to render it on the server

That's already the case?

> I think that the usage of this component should be limited to the main elements of the page

Are you sure search functionality is not a main element for a landing page? Could you name the main elements of the landing page for future reference?

Reply:
> That's already the case?

The initial render is performed without the search box: https://github.com/mui-org/material-ui/blob/dbc1aee89b2a50cf248683639e490caf1202f4b0/docs/src/modules/components/AppFrame.js#L70

> Are you sure search functionality is not a main element for a landing page?

The analytics suggests that the most common search path is Google. If we take last week as an example:
- Google: 305k searches
- Algolia: 193k searches

> Could you name the main elements of the landing page for future reference?

I agree that if we only take the landing page into consideration, the concern is different :). I wonder if always displaying the search, server-side even if not interactive could be better, not sure 🤔

Reply:
Benchmark with Youtube, the search bar isn't skeletoned nor server-side rendered. Maybe what we had before was already our best option :)

Reply:
> The server-side render is performed without the search box:

What is the issue here? We're already not rendering the search on the server which you said you'd prefer. 

> >Are you sure search functionality is not a main element for a landing page?

> If we take all the pages into account, the analytics suggest that the most common search path is Google. 

That answers not the question you posed. What are the main/most used elements on the landing page? You seemed to know this and I'd like to have this knowledge as well so that I can avoid work on unused parts of the docs.

Reply:
> What is the issue here? 

I think that we should be concerned with:

1. The visual distraction it creates on a component page, moving attention away from the body of the page. Removing the fast blink (a), the color jump (b) or the layout jump (c) are leverages we can use to reduce this concern. Actually (a) could be a nice feature to build into the component. For (b), we would need a custom background color. For (c) we would need to apply the same margins.
2. When the search box should start to be visible/interactive? Given that we have no server-side support for the search https://github.com/algolia/docsearch/issues/769, I wonder if we don't already have the best answer to this problem. Not sure.

> What are the main/most used elements on the landing page?

I agree that, on the landing page, the search is important. I would imagine the most used elements are 1. the get started button, 2. the drawer, 3. the search. Now, the handling of the search for the landing page vs the other pages is identical. So either we should consider the situation as a whole or introduce different answers to the problem per context.

---

What do you think of removing the usage of the skeleton?

Reply:
> I would imagine

So you don't actually know. Would've been easier if you started with that instead of beating around the bush.

> Removing the fast blink

Removed the fallback from the server rendered output. Suspense will handle the heavy lifting regarding display duration.

> color jump

So you think the Skeleton is a useless component anyway? There will always be a color jump. Otherwise you couldn't use a Skeleton

> layout jump

Fixed

>  I wonder if we don't already have the best answer to this problem. Not sure.

I have no idea what you're talking about. Could you explain what issue you have? The search is not rendered on the server. It is still not rendered on the server in this PR. I have no plans to change this. You agree. Why is it then we need to talk about it here?

Reply:
> So you don't actually know. Would've been easier if you started with that instead of beating around the bush.

I don't think that it's Manichaean, while we don't have actual metrics on what developers are using on the landing page, we have a couple of weak signals (our own experience, what seems logical, etc.) we can leverage to make an educated guess.